tests/unit/s2n_server_extensions_test.c

```diff
@@ -56,6 +56,73 @@ int main(int argc, char **argv)
     EXPECT_SUCCESS(s2n_test_cert_chain_and_key_new(&chain_and_key,
             S2N_DEFAULT_TEST_CERT_CHAIN, S2N_DEFAULT_TEST_PRIVATE_KEY));
 
```

Comment:
we probably should have recv tests here for tls12, and have them ran again with s2n_enable_tls13() but `actual_protocol_version = S2N_TLS12`